CMakeLists.txt

```diff
@@ -17,22 +17,31 @@ set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake" ${CMAKE_MODULE_PATH})
 set(CMAKE_CXX_FLAGS_RELEASE "-O3 -funroll-loops -ftree-vectorize")
 set(CMAKE_FIND_PACKAGE_PREFER_CONFIG TRUE)
 
+set(COMPILE_OPTIONS -D_REENTRANT -fPIC)
+
 # Require PkgConfig
 find_package(PkgConfig REQUIRED)
 find_package(OpenMP REQUIRED)
 
-# TBB CMake integration
-message(STATUS "Using math TBB")
-# Define TBB_ROOT Folder
-set(TBB_ROOT ${CMAKE_CURRENT_LIST_DIR}/lib/math/lib/tbb)
-file(COPY ${CMAKE_CURRENT_LIST_DIR}/lib/math/lib/tbb_2019_U8/ DESTINATION ${TBB_ROOT})
-# Build TBB Library with CMake Integration
-include(${TBB_ROOT}/cmake/TBBBuild.cmake)
-list(APPEND MAKE_ARGS "tbb_build_dir=${TBB_ROOT}")
-list(APPEND MAKE_ARGS "tbb_build_prefix=tbb")
-tbb_build(TBB_ROOT ${TBB_ROOT} CONFIG_DIR TBB_DIR MAKE_ARGS ${MAKE_ARGS})
-# Require TBB library (for compile / link options)
-find_package(TBB)
+find_package(TBB 2021)
```

Comment:
Just a comment I would like to share: at this stage, if one has TBB 2020.3 installed, bayesmix goes with the Math TBB. What if we rely on system TBB >=2020? I guess it is just a simple change in this file, but maybe it is unnecessary. What do you think?